Wrappers/Python/cil/optimisation/functions/IndicatorBox.py

```diff
@@ -29,67 +35,389 @@ class IndicatorBox(Function):
+        self.orig_lower = lower
+        self.orig_upper = upper
+        # default is to evaluate the function
+        self._suppress_evaluation = False
+
+    @property
+    def suppress_evaluation(self):
+        return self._suppress_evaluation
+
+    def set_suppress_evaluation(self, value):
+        '''Suppresses the evaluation of the function
         
-        r'''Proximal operator of IndicatorBox at x
+        Parameters
+        ----------
 
-            .. math:: prox_{\tau * f}(x)
+            value : bool
+                If True, the function evaluation on any input will return 0, without calculation.
         '''
+        if not isinstance(value, bool):
+            raise ValueError('Value must be boolean')
+        self._suppress_evaluation = value
+
+    def __call__(self,x):
+        '''Evaluates IndicatorBox at x
         
-        if out is None:
-            return (x.maximum(self.lower)).minimum(self.upper)        
-        else:               
-            x.maximum(self.lower, out=out)
-            out.minimum(self.upper, out=out) 
+        Parameters
+        ----------
+        
+            x : DataContainer
             
+        Evaluates the IndicatorBox at x. If ``suppress_evaluation`` is ``True``, returns 0.  
+        '''
+        if not self.suppress_evaluation:
+            return self.evaluate(x)    
+        return 0.0
+    
     def proximal_conjugate(self, x, tau, out=None):
         
         r'''Proximal operator of the convex conjugate of IndicatorBox at x:
 
-          ..math:: prox_{\tau * f^{*}}(x)
+          .. math:: prox_{\tau * f^{*}}(x)
+
+          Parameters
+          ----------
+
+            x : DataContainer
+                Input to the proximal operator
+            tau : float
+                Step size. Notice it is ignored in IndicatorBox, see ``proximal`` for details
+            out : DataContainer, optional
+                Output of the proximal operator. If not provided, a new DataContainer is created.
+
         '''
 
+        # x - tau * self.proximal(x/tau, tau)
+        should_return = False
+        
+        if out is None:
+            out = self.proximal(x, tau)
+            should_return = True
+        else:
+            self.proximal(x, tau, out=out)
+        
+        out.sapyb(-1., x, 1., out=out)
+
+        if should_return:
+            return out
+
+    def proximal(self, x, tau, out=None):
+        
+        r'''Proximal operator of IndicatorBox at x
+
+        .. math:: prox_{\tau * f}(x)
+
+        Parameters
+        ----------
+
+        x : DataContainer
+            Input to the proximal operator
+        tau : float
+            Step size. Notice it is ignored in IndicatorBox
+        out : DataContainer, optional
+            Output of the proximal operator. If not provided, a new DataContainer is created.
+
+        Note
+        ----
+
+            ``tau`` is ignored but it is in the signature of the generic Function class
```

Comment:
indent levels

Reply:
this is required to render as a note box.